pallets/ddc-accounts/src/lib.rs

```diff
@@ -0,0 +1,565 @@
+	pub fn DefaultBucketCount<T: Config>() -> u128 {
+		0_u128
+	}
+	#[pallet::storage]
+	#[pallet::getter(fn buckets_count)]
+	pub type BucketsCount<T: Config> =
+		StorageValue<Value = u128, QueryKind = ValueQuery, OnEmpty = DefaultBucketCount<T>>;
+
+	/// Map from bucket ID to to the bucket structure
+	#[pallet::storage]
+	#[pallet::getter(fn buckets)]
+	pub type Buckets<T: Config> =
+		StorageMap<_, Twox64Concat, u128, Bucket<T::AccountId>, OptionQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// An account has bonded this amount. \[stash, amount\]
+		///
+		/// NOTE: This event is only emitted when funds are bonded via a dispatchable. Notably,
+		/// it will not be emitted for staking rewards when they are added to stake.
+		Deposited(T::AccountId, BalanceOf<T>),
+		/// An account has unbonded this amount. \[stash, amount\]
+		Unbonded(T::AccountId, BalanceOf<T>),
+		/// An account has called `withdraw_unbonded` and removed unbonding chunks worth `Balance`
+		/// from the unlocking queue. \[stash, amount\]
+		Withdrawn(T::AccountId, BalanceOf<T>),
+		/// Total amount charged from all accounts to pay CDN nodes
+		Charged(BalanceOf<T>),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Not a controller account.
+		NotController,
+		/// Not a stash account.
+		NotStash,
+		/// Stash is already bonded.
+		AlreadyBonded,
+		/// Controller is already paired.
+		AlreadyPaired,
+		/// Cannot deposit dust
+		InsufficientDeposit,
+		/// Can not schedule more unlock chunks.
+		NoMoreChunks,
+		/// Internal state has become somehow corrupted and the operation cannot continue.
+		BadState,
+		/// Current era not set during runtime
+		DDCEraNotSet,
+	}
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig;
+
+	#[cfg(feature = "std")]
+	impl Default for GenesisConfig {
+		fn default() -> Self {
+			Self
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+			let account_id = <Pallet<T>>::account_id();
+			let min = <T as pallet::Config>::Currency::minimum_balance();
+			if <T as pallet::Config>::Currency::free_balance(&account_id) < min {
+				let _ = <T as pallet::Config>::Currency::make_free_balance_be(&account_id, min);
+			}
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Create new bucket with provided public availability & reserved resources
+		///
+		/// Anyone can create a bucket
+		#[pallet::weight(10_000)]
+		pub fn create_bucket(
+			origin: OriginFor<T>,
+			public_availability: bool,
+			resources_reserved: u128,
+		) -> DispatchResult {
+			let bucket_owner = ensure_signed(origin)?;
+			let cur_bucket_id = Self::buckets_count();
+
+			let bucket = Bucket {
+				bucket_id: cur_bucket_id + 1,
+				owner_id: bucket_owner,
+				public_availability,
+				resources_reserved,
+			};
+
+			<BucketsCount<T>>::set(cur_bucket_id + 1);
+			<Buckets<T>>::insert(cur_bucket_id + 1, bucket);
```

Comment:
`cur_bucket_id + 1` is repeated 3 times. Let's increment it here:
```
let cur_bucket_id = Self::buckets_count() + 1;
```

Reply:
fixed